### ISSUE TITLE
[14.0] Base automation permission issue

### ISSUE THREAD

**Issue by litnimax** [odoo/odoo#59680]
Fresh Odoo 14.0 without any customization.
Error happends when trying to send an email when new forum post is created.
So I created a 'New forum post' automation rule with Action To Do = Send email.
When trying to create a new forum post there is an error:
```
odoo_1  | 2020-10-09 20:24:49,585 14 INFO 14_odoopbx.com odoo.addons.base.models.ir_model: Access Denied by ACLs for operation: read, uid: 16
, model: base.automation
odoo_1  | 2020-10-09 20:24:49,589 14 INFO 14_odoopbx.com odoo.addons.base.models.ir_model: Access Denied by ACLs for operation: read, uid: 16
, model: base.automation
odoo_1  | 2020-10-09 20:24:49,592 14 WARNING 14_odoopbx.com odoo.addons.http_routing.models.ir_http: 403 Forbidden:
odoo_1  |
odoo_1  | Traceback (most recent call last):
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
odoo_1  |     result = request.dispatch()
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 805, in dispatch
odoo_1  |     r = self._call_function(**self.params)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 358, in _call_function
odoo_1  |     return checked_call(self.db, *args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 94, in wrapper
odoo_1  |     return f(dbname, *args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 346, in checked_call
odoo_1  |     result = self.endpoint(*a, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 911, in __call__
odoo_1  |     return self.method(*args, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 530, in response_wrap
odoo_1  |     response = f(*args, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/website_forum/controllers/main.py", line 350, in post_create
odoo_1  |     'tag_ids': post_tag_ids
odoo_1  |   File "<decorator-gen-186>", line 2, in create
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/api.py", line 344, in _model_create_multi
odoo_1  |     return create(self, [arg])
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 310, in create
odoo_1  |     action._process(action._filter_post(records))
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 204, in _filter_post
odoo_1  |     return self._filter_post_export_domain(records)[0]
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 208, in _filter_post_export_domain
odoo_1  |     if self.filter_domain and records:
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/fields.py", line 952, in __get__
odoo_1  |     record._fetch_field(self)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3050, in _fetch_field
odoo_1  |     self._read(fnames)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3064, in _read
odoo_1  |     self.check_access_rights('read')
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3312, in check_access_rights
odoo_1  |     return self.env['ir.model.access'].check(self._name, operation, raise_exception)
odoo_1  |   File "<decorator-gen-33>", line 2, in check
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 90, in lookup
odoo_1  |     value = d[key] = self.method(*args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_model.py", line 1759, in check
odoo_1  |     raise AccessError(msg)
odoo_1  | Exception
odoo_1  |
odoo_1  | The above exception was the direct cause of the following exception:
odoo_1  |
odoo_1  | Traceback (most recent call last):
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
odoo_1  |     result = request.dispatch()
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 805, in dispatch
odoo_1  |     r = self._call_function(**self.params)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 358, in _call_function
odoo_1  |     return checked_call(self.db, *args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 94, in wrapper
odoo_1  |     return f(dbname, *args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 346, in checked_call
odoo_1  |     result = self.endpoint(*a, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 911, in __call__
odoo_1  |     return self.method(*args, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/http.py", line 530, in response_wrap
odoo_1  |     response = f(*args, **kw)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/website_forum/controllers/main.py", line 350, in post_create
odoo_1  |     'tag_ids': post_tag_ids
odoo_1  |   File "<decorator-gen-186>", line 2, in create
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/api.py", line 344, in _model_create_multi
odoo_1  |     return create(self, [arg])
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 310, in create
odoo_1  |     action._process(action._filter_post(records))
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 204, in _filter_post
odoo_1  |     return self._filter_post_export_domain(records)[0]
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 208, in _filter_post_export_domain
odoo_1  |     if self.filter_domain and records:
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/fields.py", line 952, in __get__
odoo_1  |     record._fetch_field(self)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3050, in _fetch_field
odoo_1  |     self._read(fnames)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3064, in _read
odoo_1  |     self.check_access_rights('read')
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/models.py", line 3312, in check_access_rights
odoo_1  |     return self.env['ir.model.access'].check(self._name, operation, raise_exception)
odoo_1  |   File "<decorator-gen-33>", line 2, in check
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 90, in lookup
odoo_1  |     value = d[key] = self.method(*args, **kwargs)
odoo_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_model.py", line 1759, in check
odoo_1  |     raise AccessError(msg)
odoo_1  | odoo.exceptions.AccessError: You are not allowed to access 'Automated Action' (base.automation) records.
odoo_1  |
odoo_1  | This operation is allowed for the following groups:
odoo_1  |       - Administration/Settings
odoo_1  |
odoo_1  | Contact your administrator to request access if necessary.
odoo_1  |
odoo_1  | 2020-10-09 20:24:49,847 14 INFO 14_odoopbx.com werkzeug: 178.18.34.150 - - [09/Oct/2020 20:24:49] "POST /forum/support-1/new HTTP/1.1" 403 - 150 0.098 0.338
```

**Comment by sswapnesh** [comment]
Ping @mart-e 
Seems due to https://github.com/odoo/odoo/pull/53335/files#diff-e311bbb481a82f15c1102f3ce48db373L2

**Comment by litnimax** [comment]
Yes with this rule it works:
```
access_base_automation,base.automation,model_base_automation,,1,0,0,0
```

**Comment by litnimax** [comment]
But does really portal user have to have access to ALL automation rules!?
There should be another solution IMHO.

**Comment by sswapnesh** [comment]
> But does really portal user have to have access to ALL automation rules!?

I dont think so.

> There should be another solution IMHO.

Indeed, Let's wait for the feedback from @mart-e

**Comment by mart-e** [comment]
> But does really portal user have to have access to **ALL** automation rules!?

The question is even "does really portal user have to have access to **ANY** automation rules" and the answer is **no**.
The solution is probably to add `sudo` when retrieving the content of automated action.

I am working on a fix at #59740 thanks for the report
